pkg/agent/manager/cache/cache.go

```diff
@@ -556,11 +566,20 @@ func (c *Cache) notifyAll() {
 	}
 }
 
-func (c *Cache) notifyBySelectors(set selectorSet) {
-	subs, subsDone := c.getSubscribers(set)
-	defer subsDone()
-	for sub := range subs {
-		c.notify(sub)
+func (c *Cache) notifyBySelectorSet(sets ...selectorSet) {
+	notifiedSubs, notifiedSubsDone := allocSubscriberSet()
+	defer notifiedSubsDone()
+	for _, set := range sets {
+		subs, subsDone := c.getSubscribers(set)
+		defer subsDone()
+		for sub := range subs {
+			if sub.set.SuperSetOf(set) {
+				if _, notified := notifiedSubs[sub]; !notified {
+					c.notify(sub)
+					notifiedSubs[sub] = struct{}{}
+				}
+			}
```

Comment:
We can skip the superset check if we've already notified this subscriber...
```suggestion
			if _, notified := notifiedSubs[sub]; !notified && sub.set.SuperSetOf(set) {
				c.notify(sub)
				notifiedSubs[sub] = struct{}{}
			}
```